### DISCUSSION TITLE
The custom domain is not properly formatted

### DISCUSSION THREAD

**Discussion by LiriDek** [community/community#29304] [Other Feature Feedback, Questions, & Ideas] [answered]
I try to host my work on github like I did for my first projects and have a link available to share with my contacts so they can have a visual.

From the repository i go to settings, then to pages, i type a name on the "Custom Domain" part and when I save a message appears.

The custom domain "domain name" is not properly formatted.

I don't know why and i can't find a solution to the problem.

Thanks in advance for your help.

![errorgit](https://user-images.githubusercontent.com/74592475/184421422-592a732d-3c54-40f0-a2d3-8624dd565b37.PNG)

**Comment by airtower-luna** [comment, marked as answer]
The error message looks like you tried to use `eportfolio` as the custom domain. That's not a valid domain name, it's missing at least the TLD (top level domain, e.g. `.com`, `.org`, etc.) part.

**Reply by LiriDek** [reply]
i didn't thought i had to you .com or .org since i don't remember i've used one for my previous projects. Just typed a name and it was ok.

**Reply by airtower-luna** [reply]
*Domains* have to have a TLD. If you're looking to set some other kind of name it might work, but then the question is: Where do you expect that name to appear?

**Reply by LiriDek** [reply]
![first project](https://user-images.githubusercontent.com/74592475/184431789-3644fcce-c299-404e-a686-0b676ad7b550.PNG)

the name will appear on the link. 

If you look at this other project i just typed the same name as my repository without adding any TLD and it worked, that's why i'm wondering why do i have that issue now.

**Reply by LiriDek** [reply]
i also tried with a TLD and i'm having another problem 
![customdomaingit](https://user-images.githubusercontent.com/74592475/184432120-0a79441e-47e1-4b38-9c74-ff18dbf8944c.PNG)

**Reply by airtower-luna** [reply]
> the name will appear on the link.

Which part of the link do you mean? The *domain* part (`liridek.github.io` in your example), or the *path* segment (`first-github-project`)?

If you want to set the domain, you need to actually own the domain, and set up DNS correctly. The path will automatically be the same as the repository name, no need to configure anything.

**Reply by LiriDek** [reply]
i just typed first-github-project and saved it and it worked. didn't need TLD.

**Reply by airtower-luna** [reply]
I'm surprised the form didn't complain, but I can see it didn't change anything about where your site is deployed. :sweat_smile:

**Reply by LiriDek** [reply]
So what should i do to just host the website on github just like i did for that first one ?

**Reply by airtower-luna** [reply]
If you want to see the `index.html` file when you open the site URL, just move it into the root of the repository. Currently it's in a subdirectory, and deployed accordingly: https://liridek.github.io/first-github-project/Tutorial%20Frontend%20SImplified/

**Reply by LiriDek** [reply]
yeah but first of all i need my eportfolio to be hosted on github before i could have a link and watch it. That's the step where i'm blocked.

**Reply by airtower-luna** [reply]
What exactly is your "eportfolio"? If it's some document you want to show you need to upload it. If not please try to give all the relevant details of what you want to achieve, otherwise it's hard to help.

**Reply by LiriDek** [reply]
Eportfolio is my actual project of having a portfolio. I just want to host it on github.
As i told you previously, i did host other stuff on github so i could have generated link to send and to share.

How could i do it again since they do not let me do that ?

**Reply by airtower-luna** [reply]
> How could i do it again since they do not let me do that ?

Do you already have the data you want to publish in a repository? In that case please post a link *to that repository*.

**Reply by LiriDek** [reply]
Yes i do have : https://github.com/LiriDek/Tutorial-Eportfolio

**Reply by airtower-luna** [reply]
Okay, can see a few things with that:

* There is a working Pages build and deployment: https://github.com/LiriDek/Tutorial-Eportfolio/actions/runs/2849587354
* You've configured the site to use the custom domain `eportfolio.org`.
* You don't seem to actually *own* that domain, it seems to point at some domain reseller.

You have two options:

1. Disable the custom domain, remove the CNAME file, and your site will appear under the `github.io` subdomain matching your username, plus the repository name as path. So in your case: `https://lilidek.github.io/Tutorial-Eportfolio/`
2. Buy a domain and use that as your custom domain.

**Reply by LiriDek** [reply]
I want to do the 1st option. Where is the CNAME file ?
how do i stop that working page since i don't want to own a domain right now ?

**Reply by airtower-luna** [reply]
The CNAME file is right here in your repository: https://github.com/LiriDek/Tutorial-Eportfolio/blob/master/CNAME

> how do i stop that working page since i don't want to own a domain right now ?

You don't need to stop anything, just remove the custom domain configuration (including the CNAME file).

See also: https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#removing-a-custom-domain

**Reply by LiriDek** [reply]
Finally i can see a link but i don't know how i did that seriously... i just removed the CNAME from VSCode and and the custom domain than i saw a link...

I don't really understand how the link appeared.

**Reply by airtower-luna** [reply]
Maybe because without the non-working custom domain there's a working deployment link now, but again it's hard to say without seeing what exactly to mean. Anyway, if it works now (I can at least see a site) marking the question solved would be appreciated. :slightly_smiling_face:

**Reply by LiriDek** [reply]
Of course thanks for your time and your help sir really appreciated

**Reply by airtower-luna** [reply]
If you want to use a honorific it's "ma'am". :wink:

**Reply by LiriDek** [reply]
Thank you my lady :)

Now i'm trying to figure out how to classify it as solved

**Reply by airtower-luna** [reply]
There should be this thing at the top level answer.
![grafik](https://user-images.githubusercontent.com/5814919/184494206-669c674b-6dce-44b4-8f8e-ac56a4d102e0.png)

**Reply by LiriDek** [reply]
Thanks again :)

**Comment by christo-blip** [comment]
I also got the same problem I don't have any domain, can you give an example of how should the customer domain format should be

**Reply by airtower-luna** [reply]
A typical domain name looks like `www.example.com`, see the Wikipedia article on [domain names](https://en.wikipedia.org/wiki/Domain_name) if you want to know more. Mind that you'll have to register (buy) the domain you want to use before you can configure it on the GitHub side!

**Comment by hoodaatwa** [comment]
The custom domain `github.com/hoodaatwa/testfor` is not properly formatted. See https://docs.github.com/articles/troubleshooting-custom-domains/#github-repository-setup-errors for more information
Same problem here

**Reply by ryzen322** [reply]
I hope this is not too late to address your issue about uploading your site to GitHub 
go back to your code

![1](https://user-images.githubusercontent.com/66658075/217523977-26566628-eeee-473b-b738-c7c987ee672e.png)

click the check 

![image](https://user-images.githubusercontent.com/66658075/217524153-de214bf6-0957-4f25-a392-bf83210e9569.png)

click the details
select the project what you've built ago 

![image](https://user-images.githubusercontent.com/66658075/217524424-561dfca3-52ef-4b30-8be0-ae21e21b3cdf.png)

click deploy 

![image](https://user-images.githubusercontent.com/66658075/217524749-3a1c7d5f-b742-44c6-899a-e7763092ab9f.png)

there is a url that youve created ago
